control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -54,10 +57,41 @@ type GatewayController struct {
 	client.Client
 }
 
+func buildOpts(ref api.ConfigEntry) cmp.Option {
+	switch v := ref.(type) {
```

Comment:
I think all of these can technically just use a path-based comparison without doing the struct reflection that requires this switch, but I think it's fine to leave as-is for now.